### PR TITLE
Use URI scheme

### PR DESCRIPTION
**Pull request by wagoodman**
Today we allow for the following user input strings:
```
docker://alpine:latest
docker://anchore/anchore-engine
ubuntu:latest
docker-archive://some/path/to.tar
tar://some/path/to.tar
tarball://some/path/to.tar
oci-directory://some/path/to/dir
oci-archive://some/path/to.tar
oci-tar://some/path/to.tar
```

This PR aims to bring us closer to the URI specifications (RFC2396 and RFC3986) by eliminating the URL-specific authority indicator (`//`) from the string requests and use purely a scheme approach:

```
docker:alpine:latest
docker:anchore/anchore-engine
ubuntu:latest
docker-archive:some/path/to.tar
oci-archive:some/path/to.tar

# new: automatically detect the image... this will automatically resolve to either oci-directory, oci-archive, or docker-archive
./just/some/path
```

We also want to remove redundant options:
- tar
- tarball
- oci-directory
- oci
- docker-tar

This is to minimize the risk of a scheme overlap with user input.

Lastly, we came to a conclusion from offline discussions that we should migrate image fixture utils from the `go-testutils` repo to here, as currently there is a cyclic package dependency problem which complicates updating the feature set as a whole. These utils have been moved to `pkg/imagetest` (this was a good PR to do so since a change to the utility functions was required for the above scheme changes).

Partially addresses: https://github.com/anchore/syft/issues/177 and https://github.com/anchore/grype/issues/157

### CODE REVIEW

**Comment by wagoodman** [comment]
@luhring @alfredodeza 

Though looking closer at the `docker:` scheme, it may make sense to leave this with an authority component, since it may imply that the image reference is relative to the docker daemon to interpret the remaining path. The same would be true for `podman` when that support is added.

That is, I think leaving `docker://` as is since we are querying the [central] docker daemon and it is interpreting the path elements --the docker daemon is acting as the authority. This also clears up some edge cases outlined in the tests (e.g. `docker:latest` vs `docker:docker:latest`).

File: pkg/image/source.go [hunk 1]
```diff
@@ -38,63 +42,82 @@ var AllSources = []Source{
...
 }
 
-// DetectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive.
+// DetectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive for a given filesystem.
 func DetectSourceFromPath(imgPath string) (Source, error) {
 	return detectSourceFromPath(afero.NewOsFs(), imgPath)
 }
 
-// detectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive for a given filesystem.
+// DetectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive for a given filesystem.
```
Comment (wagoodman) [inline pkg/image/source.go:120]: leave lowercase

File: pkg/image/source.go [hunk 2]
```diff
@@ -38,63 +42,82 @@ var AllSources = []Source{
...
+		sourceHint = candidates[0]
+		location = strings.TrimPrefix(userInput, sourceHint+SchemeSeparator)
+		source = ParseSourceScheme(sourceHint)
 	default:
 		source = UnknownSource
 	}
 
 	if source == UnknownSource {
-		return source, "", nil
+		if couldBeDockerReference(userInput) {
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply pkg/image/source.go:104]: just tested this out --it does not!

**Comment by wagoodman** [comment]
After some discussion internally, we're sticking to purely URI schemes (no authority will be assumed/needed)

**Review by luhring — APPROVED** [review]
